app/features/settings/contact.tsx

```diff
@@ -0,0 +1,107 @@
+import { ArrowUpRight, Share, Trash2 } from 'lucide-react';
+import { useNavigate, useRouteLoaderData } from 'react-router';
+import { useCopyToClipboard } from 'usehooks-ts';
+import { PageContent } from '~/components/page';
+import { Button } from '~/components/ui/button';
+import { useToast } from '~/hooks/use-toast';
+import { canShare, shareContent } from '~/lib/share';
+import { ContactAvatar } from '../contacts/contact-avatar';
+import { useContact, useDeleteContact } from '../contacts/contact-hooks';
+import { SettingsViewHeader } from './ui/settings-view-header';
+
+export function SingleContact({ contactId }: { contactId: string }) {
+  const { origin } = useRouteLoaderData('root') as { origin: string };
```

Comment:
I would maybe just create a custom hook that returns this origina/domain data so we don't have to repeat this everywhere

Reply:
yeah both is fine